File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        # so again I will iterate and this time I need to save index tho, and update everytime I find a new index ig?
        
        query = {}
        for i,n in enumerate(nums):
            if n in query:
                if i-query[n]<= k:
                    return True
                else:
                    query[n]=i
            else:
                query[n]=i
        return False

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        // ok so I need to iterate and update my closest index
        unordered_map<int,int> query;
        int n = nums.size();
        for(int i=0; i<n; i++){
            if(query.find(nums[i])!=query.end()){
                if(i-query[nums[i]]<=k){
                    return true;
                }else{
                    query[nums[i]]=i;
                }
            }else{
                query[nums[i]]=i;
            }
        }
        return false;
    }
};